src/main/java/org/roaringbitmap/Container.java

```diff
@@ -122,6 +122,16 @@ public int andCardinality(Container x) {
     }
   }
 
+  /**
+   * Returns the cardinality of the XOR between the passed container and this
+   * container without materialising a temporary container.
+   * @param other other container
+   * @return the cardinality of the symmetric difference of the two containers
+   */
+  public int xorCardinality(Container other) {
+    return getCardinality() + other.getCardinality() - 2 * andCardinality(other);

```

Comment:
Actually this is how I started off quite some time ago with the XOR cardinality pull request. The code is very complicated consisting of 6 separate container level XOR algorithms, mandating a few code duplications and is probably full of the kind of bugs you find 1 year after release. The venn diagram identity has the benefit of being obviously correct, so long as you can get andCardinality right, without much overhead as you note.